### PR TITLE
implemented tensnorflow avg_pool1d

### LINKED ISSUE #11775: avg_pool1d

### PR DESCRIPTION
**Pull request by ajithpious**
Closes #11775

### CODE REVIEW

**Comment by ajithpious** [comment]
@kurshakuz could you please review my pr.

**Review by kurshakuz — CHANGES_REQUESTED** [review]
Hey! Thank you for your contribution!

Your implementation currently fails for one of the tests:
```
E           AssertionError:  the results from backend torch and ground truth framework tensorflow do not match
E            [[[-1.        ]
E             [-0.33333334]
E             [ 0.33333334]]]!=[[[-1.        ]
E             [-0.33333334]
E             [ 0.        ]]] 
E           
E           
E           Falsifying example: test_tensorflow_avg_pool1d(
E               data_format='NWC',
E               x_k_s_p=(['float32'], [array([[[-1.],
E                         [-1.],
E                         [ 1.]]], dtype=float32)], (3,), (1,), 'SAME'),
E               test_flags=FrontendFunctionTestFlags(
E                   num_positional_args=0,
E                   with_out=False,
E                   inplace=False,
E                   as_variable=[False],
E                   native_arrays=[False],
E               ),
E               fn_tree='ivy.functional.frontends.tensorflow.nn.avg_pool1d',
E               frontend='tensorflow',
E               on_device='cpu',
E           )
E           
E           You can reproduce this example by temporarily adding @reproduce_failure('6.70.0', b'AXicY2BiAAJGBghgRGbDGCA2E4QDAAEvAAw=') as a decorator on your test case

ivy_tests/test_ivy/helpers/assertions.py:53: AssertionError
```

**Comment by ajithpious** [comment]
> Hey! Thank you for your contribution!
> 
> Your implementation currently fails for one of the tests:
> 
> ```
> E           AssertionError:  the results from backend torch and ground truth framework tensorflow do not match
> E            [[[-1.        ]
> E             [-0.33333334]
> E             [ 0.33333334]]]!=[[[-1.        ]
> E             [-0.33333334]
> E             [ 0.        ]]] 
> E           
> E           
> E           Falsifying example: test_tensorflow_avg_pool1d(
> E               data_format='NWC',
> E               x_k_s_p=(['float32'], [array([[[-1.],
> E                         [-1.],
> E                         [ 1.]]], dtype=float32)], (3,), (1,), 'SAME'),
> E               test_flags=FrontendFunctionTestFlags(
> E                   num_positional_args=0,
> E                   with_out=False,
> E                   inplace=False,
> E                   as_variable=[False],
> E                   native_arrays=[False],
> E               ),
> E               fn_tree='ivy.functional.frontends.tensorflow.nn.avg_pool1d',
> E               frontend='tensorflow',
> E               on_device='cpu',
> E           )
> E           
> E           You can reproduce this example by temporarily adding @reproduce_failure('6.70.0', b'AXicY2BiAAJGBghgRGbDGCA2E4QDAAEvAAw=') as a decorator on your test case
> 
> ivy_tests/test_ivy/helpers/assertions.py:53: AssertionError
> ```

Hi @kurshakuz, the tensorflow  avg_pool1d frontend function uses the functional api implemented in ivy\functional\ivy\experimental\layers.py . I tested the same API with different frontends and the results are different. I think this is causing the frontend to fail.
Code to repro:
a = [[[-1.], [-1.], [1.]]]
xnp = np.array(a)
xtf = tf.Variable(a)
xtch = torch.tensor(a)
print(ivy.avg_pool1d(xtf, 3, 1, "SAME"))
print(ivy.avg_pool1d(xtch,3,1,"SAME"))
print(ivy.avg_pool1d(xtf,3,1,"SAME"))

The outputs are different for the same input. Could you please have a look into this.

**Review by kurshakuz — CHANGES_REQUESTED** [review]
You are correct, outputs should not be different. I would sugges you to raise your issue in the corresponding Discord channel in case someone faced the same problem.

I also tried to run your implementation locally, and it seems to be producing incorrect results.

```
E           AssertionError:  the results from backend torch and ground truth framework tensorflow do not match
E            [[[-1.       ]
E             [-1.       ]
E             [-1.       ]]
E           
E            [[-1.       ]
E             [-1.3333334]
E             [-1.6666666]]]!=[[[-1.       ]
E             [-1.       ]
E             [-1.       ]]
E           
E            [[-1.       ]
E             [-1.3333334]
E             [-1.5      ]]] 

```

**Comment by ajithpious** [comment]
Hi @kurshakuz, the issue has been fixed in one of the commits and all the test cases are passing.

File: ivy_tests/test_ivy/test_frontends/test_tensorflow/test_nn.py [hunk 1]
```diff
@@ -1352,3 +1352,34 @@ def test_tensorflow_avg_pool(
         padding=padding,
         data_format=data_format,
     )
+
+
+@handle_frontend_test(
+    fn_tree="tensorflow.nn.avg_pool1d",
+    data_format=df(data_format=st.sampled_from(["NWC"])),
```
Comment (kurshakuz) [inline ivy_tests/test_ivy/test_frontends/test_tensorflow/test_nn.py:1359]:
```
NameError: name 'df' is not defined
```

**Comment by ivy-seed** [comment]
This PR has been labelled as stale because it has been inactive for more than 7 days. If you would like to continue working on this PR, then please add another comment or this PR will be closed in 7 days.

**Comment by ivy-seed** [comment]
This PR has been closed because it has been marked as stale for more than 7 days with no activity.

**Pull request closed without merging**